staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509_test.go

```diff
@@ -848,6 +849,238 @@ func TestX509Verifier(t *testing.T) {
 	}
 }
 
+func TestConvertToUserInfo(t *testing.T) {
+	testCases := map[string]struct {
+		input          pkix.Name
+		expectUserInfo *user.DefaultInfo
+		shouldSucceed  bool
+	}{
+		"empty CN": {
+			input: pkix.Name{
+				CommonName: "",
+				Organization: []string{
+					"xyz",
+				},
+			},
+			expectUserInfo: nil,
+			shouldSucceed:  false,
+		},
+		"old format, has tenant in CN": {
+			input: pkix.Name{
+				CommonName: "abc:def",
+				Organization: []string{
+					"xyz",
+				},
+			},
+			expectUserInfo: &user.DefaultInfo{
+				Name: "abc:def",
+				Groups: []string{
+					"xyz",
+				},
+				Tenant: "abc",
+			},
+			shouldSucceed: true,
+		},
+		"old format, no tenant but has colon in CN": {
+			input: pkix.Name{
+				CommonName: "abc:",
+				Organization: []string{
+					"xyz",
+				},
+			},
+			expectUserInfo: &user.DefaultInfo{
+				Name: "abc:",
+				Groups: []string{
+					"xyz",
+				},
+				Tenant: "abc",
+			},
+			shouldSucceed: true,
+		},
+		"old format, no tenant and no colon in CN": {
+			input: pkix.Name{
+				CommonName: "abc",
+				Organization: []string{
+					"xyz",
+				},
+			},
+			expectUserInfo: &user.DefaultInfo{
+				Name: "abc",
+				Groups: []string{
+					"xyz",
+				},
+				Tenant: "",
+			},
+			shouldSucceed: true,
+		},
+		"old format, multiple colons in CN": {
+			input: pkix.Name{
+				CommonName: "abc:def:ghi",
+				Organization: []string{
+					"xyz",
+				},
+			},
+			expectUserInfo: &user.DefaultInfo{
+				Name: "abc:def:ghi",
+				Groups: []string{
+					"xyz",
+				},
+				Tenant: "abc",
+			},
+			shouldSucceed: true,
+		},
+		"old format, nothing but a colon in CN": {
+			input: pkix.Name{
```

Comment:
Do we need to support such a CN, just a ":"?

Reply:
I prefer to not add any rule as to what should or shouldn't be in a CN.